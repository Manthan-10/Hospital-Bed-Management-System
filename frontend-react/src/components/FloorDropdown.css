.floors{
    flex-direction: column;
    background-color:white;
    /* align-items: center; */
    padding: 20px;
    /* margin-left:270px;
    margin-right:15px;
    margin-top:15px; */
    justify-content:flex-end;
}

