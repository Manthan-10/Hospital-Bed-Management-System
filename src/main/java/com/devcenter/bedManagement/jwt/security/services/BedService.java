package com.devcenter.bedManagement.jwt.security.services;

import java.util.List;
import java.util.Optional;

import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.devcenter.bedManagement.jwt.models.Bed;
import com.devcenter.bedManagement.jwt.models.Room;
import com.devcenter.bedManagement.jwt.repository.BedRepository;
import com.devcenter.bedManagement.jwt.repository.RoomRepository;


/**
 * Service class
 * @author MN097789
 *
 */

@Service
public class BedService {
	
	@Autowired
	BedRepository bedRepository;
	
	@Autowired
	RoomRepository roomRepository;
	
	@Autowired
	private ModelMapper modelMapper;
	
	/**
	 * Function for adding new beds
	 * @param bed represents Bed object
	 * 	 
	 **/
	public void createBed(Bed bed) {
		 bedRepository.save(bed);
	}
	
	

	/**
	 * Function fetches all beds
	 * @return Returns List of all beds
	 */
	public List<Bed> getBeds() {
		return bedRepository.findAll(); 
	}
	
	
	/**
	 * Function fetches the details of all the beds for the given Room id 
	 * @param is id, which represents Room Id
	 * @return All Beds in that particular Room and its details
	 * 
	 */
	public List<Bed> getBedsByRoom(int id){
		
		Optional<Room> room = roomRepository.findById(id);
		List<Bed> beds = bedRepository.getBedsByRoom(this.modelMapper.map(room, Room.class));
		
		return beds;
		
	}
}
