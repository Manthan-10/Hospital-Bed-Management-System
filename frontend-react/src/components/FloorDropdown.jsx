import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { useNavigate } from "react-router-dom";
import BedService from "../services/bed.service";
import "./FloorDropdown.css";

/**
 * This component displays all the floors in a particular Hospital
 * in a dropdown
 * @param {hid} HospitalId of that Hospital
 */
const FloorDropdown = ({ hid }) => {
  const [data, setFloorList] = useState([]);
  const { t } = useTranslation();

  /**
   * @param {hid} HospitalId of that Hospital
   * @return Floor Details of that hospital
   */
  useEffect(() => {
    BedService.getFloorsByHid(hid)
      .then((response) => {
        console.log(response);
        setFloorList(response.data);
      })
      .catch((e) => {
        console.log(e);
      });
  }, []);

  let navigate = useNavigate();

  /**
   * Handles Choosing of floors in a dropdown
   */
  function handleChange(value) {
    navigate(`/hospitals/${hid}/floor/${value}`);
    window.location.reload(false);
  }

  return (
    <div className="floors">
      <select
        className="select-container"
        size="large"
        onChange={(event) => handleChange(event.target.value)}
      >
        <option>{t("Select_Floor")} </option>
        {data.map((floor) => (
          <option key={floor.id} value={floor.id}>
            {floor.floorNo}
          </option>
        ))}
      </select>
    </div>
  );
};

export default FloorDropdown;
