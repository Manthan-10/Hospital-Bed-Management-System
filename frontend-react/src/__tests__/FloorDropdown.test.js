import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import FloorDropdown from "../components/FloorDropdown";
import Adapter from "enzyme-adapter-react-16";
import * as reactRouterDom from "react-router-dom";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import BedService from "../services/bed.service";

Enzyme.configure({ adapter: new Adapter() });

/**
 * Testing for UI of Login page
 * @author MN097789
 */

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useSelector: jest.fn(),

  useNavigate: () => mockNavigate,
}));
const mockNavigateSpy = jest.spyOn(reactRouterDom, "useNavigate");

describe("FloorDropdown UI Test Suite", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FloorDropdown />);
  });

  test("Checks if it renders correctly", () => {
    const tree = renderer.create(<FloorDropdown />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should check for divs", () => {
    wrapper = shallow(<FloorDropdown />);
    expect(wrapper.length).toEqual(1);
  });

  test("BedService.getFloorsByHid is called", () => {
    jest.spyOn(React, "useEffect").mockImplementation((f) => f());
    jest.spyOn(BedService, "getFloorsByHid");
    jest.spyOn(global.console, "log");
    render(<FloorDropdown />);
    expect(BedService.getFloorsByHid).toHaveBeenCalled();
  });
});
