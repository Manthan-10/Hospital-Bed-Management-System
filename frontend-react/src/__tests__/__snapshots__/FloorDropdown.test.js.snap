// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloorDropdown UI Test Suite Checks if it renders correctly 1`] = `
<div
  className="floors"
>
  <select
    className="select-container"
    onChange={[Function]}
    size="large"
  >
    <option>
      Select_Floor
       
    </option>
  </select>
</div>
`;
